examples/decoding/plot_receptive_field.py

```diff
@@ -0,0 +1,156 @@
+from scipy.io import loadmat
+from scipy.stats import pearsonr
+
+import mne
+from mne.decoding import ReceptiveField
+from sklearn.linear_model import Ridge
+from sklearn.model_selection import KFold
+from sklearn.preprocessing import scale
+
+
+###############################################################################
+# Load the data from the publication
+# ----------------------------------
+#
+# First we will load the data collected in [1]_. In this experiment subjects
+# listened to natural speech. Raw EEG and the speech stimulus was collected.
+# We will load these below, downsampling the data in order to speed up
+# computation since we know that our features are primarily low-frequency in
+# nature. Then we'll visualize both the EEG and speech envelope.
+
+path = mne.datasets.mtrf.data_path()
+n_decim = 2
+data = loadmat(path + '/speech_data.mat')
+raw = data['EEG'].T
+speech = data['envelope'].T
+sfreq = float(data['Fs'])
+sfreq /= n_decim
+speech = mne.filter.resample(speech, down=n_decim, npad='auto')
+raw = mne.filter.resample(raw, down=n_decim, npad='auto')
+
+
+# Read in channel positions and create our MNE objects from the raw data
+montage = mne.channels.read_montage('biosemi128')
+montage.selection = montage.selection[:128]
+info = mne.create_info(montage.ch_names[:128], sfreq, 'eeg', montage=montage)
+raw = mne.io.RawArray(raw, info)
+n_channels = len(raw.ch_names)
+
+# Plot a sample of brain and stimulus activity
+fig, ax = plt.subplots()
+ax.plot(scale(raw[:][0][..., :800].T), color='k', alpha=.1)
+ax.plot(scale(speech[0, :800]), color='r', lw=2)
+ax.set(title="Sample brain and stimulus activity", xlabel="Time (s)")
+mne.viz.tight_layout()
+
+###############################################################################
+# Create and fit a receptive field model
+# --------------------------------------
+#
+# We will construct a model to find the linear relationship between the EEG
+# signal and a time-delayed version of the speech envelope. This allows
+# us to make predictions about the response to new stimuli.
+
+# Define the delays that we will use in the receptive field
+tmin, tmax = -.4, .2
+
+# Initialize the model
+spec_names = ['freq_%s' % ii for ii in range(speech.shape[0])]
+rf = ReceptiveField(tmin, tmax, sfreq,
+                    feature_names=spec_names, model=Ridge(alpha=1.))
+n_delays = int((tmax - tmin) * sfreq) + 2  # +2 to account for 0 + end
+
+n_splits = 3
+cv = KFold(n_splits)
+
+# Prepare model data
+Y, _ = raw[:]  # Outputs for the model
+Y = np.rollaxis(Y, -1, 0)  # Make time the first dimension
+speech = np.rollaxis(speech, -1, 0)
+
+# Iterate through folds, fit the model, and predict/test on held-out data
+coefs = np.zeros([n_splits, n_channels, n_delays])
+scores = np.zeros([n_splits, n_channels])
+for ii, (train, test) in enumerate(cv.split(speech)):
+    print('CV iteration %s' % ii)
+    rf.fit(speech[train], Y[train])
+    preds = rf.predict(speech[test])
```

Comment:
that's a fair point - so what about fitting then? Right now it throws away points that don't have all the time-lags covered. That OK?

on the prediction side, I'm fine w/ having `predict` use all the data. I'll still create a `mask_predict_` attribute in case users want it after-the fact. Does that work?

Then I'll add a `score` method that does the magic under the hood and I can modify the example to use this instead of manually doing it in the example. Are folks friendly to that?